recommendation/microsoft.owin.json

```diff
@@ -136,17 +136,60 @@
               "TargetCPU": [ "x86", "x64", "ARM32", "ARM64" ]
             }
           ],
-          "Description": "Add a comment to explain how to remove the base class of OwinMiddleware.",
+          "Description": "Remove the base class for owin middleware, remove the override method modifier for the invoke method, add a new gloable xpression for RequestDelegate _next, initialize it in the constructor and remove the constructor intializer base(next).",
           "Actions": [
             {
-              "Name": "AddComment",
+              "Name": "RemoveBaseClass",
+              "Type": "Class",
+              "Value": "OwinMiddleware",
+              "Description": "Remove the base class OwinMiddleware",
+              "ActionValidation": {
+                "Contains": "",
+                "NotContains": ":OwinMiddleware"
+              }
+            },
+            {
+              "Name": "ReplaceMethodModifiers",
               "Type": "Class",
-              "Value": "Please remove the base class here : OwinMiddleware",
-              "Description": "Add a comment to remove the base class OwinMiddleware",
+              "Value": {
+                "methodName": "Invoke",
+                "modifiers": "public"
+              },
+              "Description": "Replace all the invoke method modifiers with only public in order to remove the override modifier.",
               "ActionValidation": {
-                "Contains": "Pleaseremovethebaseclasshere:OwinMiddleware",
+                "Contains": "public",
+                "NotContains": "override"
+              }
+            },
+            {
+              "Name": "AddGlobalExpression",
+              "Type": "Class",
+              "Value": "RequestDelegate _next = null;",
+              "Description": "Add a new global variable to replace the OwinMiddleware global abstract variable of Next.",
+              "ActionValidation": {
+                "Contains": "RequestDelegate_next=null;",
                 "NotContains": ""
               }
+            },
+            {
+              "Name": "AddConstructorExpression",
+              "Type": "Class",
+              "Value": "_next = next;",
+              "Description": "Add a new expression in the constructor to initialize the new variable of _next to the next element in the pipeline.",
+              "ActionValidation": {
+                "Contains": "_next=next;",
+                "NotContains": ""
+              }
+            },
+            {
+              "Name": "RemoveConstructorInitializer",
+              "Type": "Class",
+              "Value": "next",
+              "Description": "Remove the base initializer for the constructor.",
+              "ActionValidation": {
+                "Contains": "",
+                "NotContains": ":base(next)"
+              }
```

Comment:
It is not but it is based on an abstract class implementation which would follow this convention. We could rename the constructor initializer parameter first and then remove it if needed but this would require a new action to be implemented just for this. Better to revisit in the future when we have need for this action and just add it.